### ISSUE TITLE
NVDA does not read the Contents Item key (Alt text) on Link annotations

### ISSUE THREAD

**Issue by FvdGDTP** [nvaccess/nvda#15229]
PDF/UA requires Annotations to have a description, in the old versions of Acrobat you could Go to the Tag Properties and fill in the Alt Text. That would place the Alt text in the Alt Text Code field in the Tag, the same place as with images read by NVDA (and Jaws).

However, PDF/UA requires it not to be ’standard’ Alt Text (like images) but in the Contents Item Key.

InDesign writes it there.

In the latest version of Acrobat Pro the Alt Text files have now split into two entries: one for images and one for Link annotations! The Link annotation Alt Text is now written also solely in the Contents Item key! The ‘image Alte Text’ field is greyed out.

However, NVDA (nor Jaws) does not read (voice) the Contents Item key where it should for Link annotations…!

So in previous versions of Acrobat we violated PDF/UA rules but things worked, now Acrobat breaks that and will not revert back because it is indeed the right correct way to put it in the Contents Item key, this won’t ‘change back’. 
So we need NVDA (and yes, Jaws) to catch up with this reality...

**Comment by Adriani90** [comment]
@FvdGDTP could you please fill the issue template properly with steps to reproduce, actual behavior and expected behavior? Please also attach an example document where we can test with NVDA on the field. This will help to investigate the issue further. Thanks in advance for your efforts.

**Comment by FvdGDTP** [comment]
Sorry, this is copied over from the mail to Quentin through the NVDA facebook group.
You can create any example yourself: create a hyperlink, use the latest version of Acrobat to enter ‘Alt text for link’ in the tag properties, then let NVDA read it and you will notice NVDA does not voice it as that ‘Alt text’ is written in the Contents item key as required by PDF/UA.
It is very simple…

**Comment by seanbudd** [comment]
Welcome @FvdGDTP ,

We are unable to process this issue as it stands.

- Please fill out the issue template fully.
- To be able to reproduce this issue, we need a sample.

Kind Regards,
NV Access Software Developers

**Comment by FvdGDTP** [comment]
Really? I mean: really? It is soooo basic…Sigh…Regards,Frans van der ***@***.*** 1 aug. 2023 om 01:52 heeft Sean Budd ***@***.***> het volgende geschreven:﻿
Welcome @FvdGDTP ,
We are unable to process this issue as it stands.

Please fill out the issue template fully.
To be able to reproduce this issue, we need a sample.

Kind Regards,
NV Access Software Developers

**Comment by FvdGDTP** [comment]
### Is your feature request related to a problem? Please describe.
Yes. NVDA does NOT read the text available in the Content items key for Link Annotations.
PDF/UA requires the 'Alt text' for (hyper)links to be presented in the Contents item key, NOT in the Alt text key (like images).
Acrobat Pro in the latest version, as well as InDesign, correctly now write 'Alt text' to this Contents item key. 
NVDA skips this entry and does not read the provided 'Alt text' in the Contents item key

### Describe the solution you'd like
NVDA should confirm to PDF/UA and read the 'Alt text' present in the Contents item key like it did with old style Alt Text for hyperlink/Link annotations

### Describe alternatives you've considered
There is none as of now Acrobat Pro only writes to the Contents item key.

### Additional context
PDF/UA requires Annotations to have a description, in the old versions of Acrobat you could Go to the Tag Properties and fill in the Alt Text. That would place the Alt text in the Alt Text Code field in the Tag, the same place as with images read by NVDA (and Jaws).
However, PDF/UA requires it not to be ’standard’ Alt Text (like images) but in the Contents Item Key.
InDesign writes it there.
In the latest version of Acrobat Pro the Alt Text files have now split into two entries: one for images and one for Link annotations! The Link annotation Alt Text is now written also solely in the Contents Item key! The ‘image Alte Text’ field is greyed out.
However, NVDA (nor Jaws) does not read (voice) the Contents Item key where it should for Link annotations…!
So in previous versions of Acrobat we violated PDF/UA rules but things worked, now Acrobat breaks that and will not revert back because it is indeed the right correct way to put it in the Contents Item key, this won’t ‘change back’.
So we need NVDA (and yes, Jaws) to catch up with this reality...

Attached screenhots of the Tag properties and a simple PDF (like you could have made yourself very easely by the way)

<img width="658" alt="Schermafbeelding 2023-07-26 at 14 10 46" src="https://github.com/nvaccess/nvda/assets/26364276/2636fbcd-5c5e-4c7d-b254-8b35232911f5">
<img width="726" alt="Schermafbeelding 2023-07-26 at 14 09 24" src="https://github.com/nvaccess/nvda/assets/26364276/c2798203-13eb-4c60-b8e4-4b27c094988d">
<img width="843" alt="Schermafbeelding 2023-08-01 om 15 08 01" src="https://github.com/nvaccess/nvda/assets/26364276/e8932f11-6418-483a-91e5-14ae7c0cd829">
<img width="869" alt="Schermafbeelding 2023-08-01 om 15 08 20" src="https://github.com/nvaccess/nvda/assets/26364276/540d2a9e-acbc-4a9e-ba10-5ca772d17f27">
[Example_fail_contents_item_read.pdf](https://github.com/nvaccess/nvda/files/12229480/Example_fail_contents_item_read.pdf)

**Comment by Adriani90** [comment]
@seanbudd why does this have an (archived) label? Is there any guidance for this specific label?

**Comment by seanbudd** [comment]
@Adriani90 - the label itself is archived, I've updated it
